src/beakerlib.sh

```diff
@@ -31,6 +31,10 @@
 #   Boston, MA 02110-1301, USA.
 #
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Command aliases for compatibilities
+declare READLINK_CMD="readlink"
+
```

Comment:
It should be enough to move all the `*_CMD` variable defines here. The individual definitions on other scripts should not be necessary unless someone would like to use them separately which is not expected.

Reply:
So all scripts source `beakerlib.sh`? Is it done recursively like the user's `test.sh` -> `beakerlib.sh` -> `logging.sh` -> `beakerlib.sh`? Or is it that the individual `.sh` scripts will not work without `*_CMD` having been defined beforehand?

What about unit-tests, no need for anything special there as well?

Reply:
> So all scripts source `beakerlib.sh`? Is it done recursively like the user's `test.sh` -> `beakerlib.sh` -> `logging.sh` -> `beakerlib.sh`? Or is it that the individual `.sh` scripts will not work without `*_CMD` having been defined beforehand?

No, the test sources `beakerlib.sh` which would define the variables so the other scripts, which are sourced by the beakerlib.sh, will have it available because they run in the same scope/namespace as they are just sourced, not executed. The individual sub-scripts, e.g. `infrastructure.sh` should not be sourced directly.

> What about unit-tests, no need for anything special there as well?

I'm wondering how it could be tested differently than just calling some of the affected functions.. The only way I can imagine is be to provide a custom bash function/script as a wrapper which would e.g. echo some additional text which could be caught for an assert. Something like `getopt_test() {echo "a getopt wrapper" >&2; getopt "$@"; } GETOPT_CMD='getopt_test'; rlHash 'test';` and then check the "a getopt wrapper" was logged on an error log level.

Reply:
I was thinking more on the lines of if the unit tests will still work, i.e. do those tests source `beakerlib.sh` or `infrastructure.sh` directly. If the later, we might need to add a `declare` in those places as well.

For writing the unit test for `getopt_test`, I am not sure I follow. That would depend on if `beakerlib.sh` is sourced, and if we can avoid the whole thing not breaking down if we re-define `GETOPT_CMD`.